cmake/kokkos_arch.cmake

```diff
@@ -104,12 +104,15 @@ IF(KOKKOS_ENABLE_COMPILER_WARNINGS)
     LIST(APPEND GNU_WARNINGS "-Wimplicit-fallthrough")
   ENDIF()
 
-  COMPILER_SPECIFIC_FLAGS(
-    COMPILER_ID CMAKE_CXX_COMPILER_ID
-    NVHPC       NO-VALUE-SPECIFIED
-    GNU         ${GNU_WARNINGS}
-    DEFAULT     ${COMMON_WARNINGS}
-  )
+  # Not using COMPILER_SPECIFIC_FLAGS function so the warning flags are not passed downstream
+  IF(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
+    STRING(REPLACE ";" " " WARNING_FLAGS "${GNU_WARNINGS}")
+  ELSEIF(CMAKE_CXX_COMPILER_ID STREQUAL NVHPC)
+    # FIXME_NVHPC
+  ELSE()
+    STRING(REPLACE ";" " " WARNING_FLAGS "${COMMON_WARNINGS}")
+  ENDIF()
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WARNING_FLAGS}")
```

Comment:
You can call `target_compiler_options(Foo ... PRIVATE)` so that they're used internally but not inherited.

This isn't an argument to make such a change here and now, though.

Reply:
Sounds good, especially that there's now a record of why.

Reply:
Setting `CMAKE_CXX_FLAGS` can break projects including kokkos as a subdirectory

Reply:
This is only if you opt-in by setting `Kokkos_ENABLE_COMPILER_WARNINGS`.  If we add as a `PRIVATE` compile-option then we have to such for every single target...

Reply:
Target options are inherited. E.g. if we set it for the Kokkos library archive, then all objects compiled to go into that will get it as well